pkg/operator/ceph/csi/util.go

```diff
@@ -169,18 +168,22 @@ func applyToPodSpec(pod *corev1.PodSpec, n *corev1.NodeAffinity, t []corev1.Tole
 	}
 }
 
-func getPortFromENV(env string, defaultPort uint16) uint16 {
-	port := os.Getenv(env)
+func getPortFromConfig(clientset kubernetes.Interface, env string, defaultPort uint16) uint16 {
+	port, err := k8sutil.GetOperatorSetting(clientset, controller.OperatorSettingConfigMapName, env, strconv.Itoa(int(defaultPort)))
+	if err != nil {
+		logger.Warningf("failed to load value for env %q. using default port %d. %v", env, defaultPort, err)
+		return defaultPort
```

Comment:
Even in this case let's return failure. We don't want to use the default port if there was just some intermittent issue querying the configmap